camel/societies/role_playing.py

```diff
@@ -256,12 +272,22 @@ def init_agents(self, init_assistant_sys_msg: BaseMessage,
             output_language (str, optional): The language to be output by the
                 agents.
         """
+        if self.assistant_functions is not None:
+            assistant_config = FuncConfig.from_openai_function_list(
+                function_list=self.assistant_functions,
+                function_call="auto",
+            )
+        else:
+            assistant_config = None
         self.assistant_agent = ChatAgent(
             init_assistant_sys_msg,
+            model_config=assistant_config,
             output_language=output_language,
+            function_list=self.assistant_functions,
```

Comment:
One last concern is do we want to `function_list` as input args? Since it is always should be always the same as the `function_list` used to define `model_config`. A better way may be to get the `function_list` from `model_config` inside `ChatAgent`. @MorphlingEd @Obs01ete @dandansamax WDYT?

Reply:
I am not sure whether I correctly understand what you mean. But I remember we previously discussed whether we need to dynamically add new functions during the execution but it seems currently we do not need this (at least we do not come up with such an application scene?). So we designed like this to fix the function list in the `model_config` object. I am wondering whether you have any suggestion on this.